src/publisher.ts

```diff
@@ -17,19 +17,22 @@
 import {promisifyAll} from '@google-cloud/promisify';
 import * as arrify from 'arrify';
 import {CallOptions} from 'google-gax';
-import {ServiceError} from 'grpc';
+import {google} from '../proto/pubsub';
 
 const each = require('async-each');
 import * as extend from 'extend';
 import * as is from 'is';
 import {Topic} from './topic';
+import {RequestCallback, Attributes} from '.';
 
-export interface PublishCallback {
-  (err: null|ServiceError, messageId: string): void;
+interface Inventory {
+  callbacks: Array<RequestCallback<string>>;
+  queued: Array<{}>;
```

Comment:
I think we could probably change this to whatever the equivalent of [PubsubMessage](https://github.com/googleapis/nodejs-pubsub/blob/master/protos/google/pubsub/v1/pubsub.proto#L384) is.

Reply:
Done!